witherable/src/Data/Witherable.hs

```diff
@@ -132,19 +136,27 @@ filterOf w f = runIdentity . filterAOf w (idDot f)
 
 -- | Remove the duplicate elements through a filter.
 ordNubOf :: Ord a => FilterLike' (State (Set.Set a)) s a -> s -> s
-ordNubOf w t = evalState (w f t) Set.empty
+ordNubOf w = ordNubOnOf w id
+
+-- | Remove the duplicate elements through a filter.
+ordNubOnOf :: Ord b => FilterLike' (State (Set.Set b)) s a -> (a -> b) -> s -> s
+ordNubOnOf w p t = evalState (w f t) Set.empty
   where
-    f a = state $ \s -> if Set.member a s
+    f a = let b = p a in state $ \s -> if Set.member b s
       then (Nothing, s)
-      else (Just a, Set.insert a s)
+      else (Just a, Set.insert b s)
 {-# INLINE ordNubOf #-}
```

Comment:
Why not use `alterF` here?

Reply:
Because I'm not even sure the `*Of` combinators should be added, `Data.Witherable` module is marked as deprecated.

Reply:
Yeah, I'm planning to remove Data.Witherable on 0.5